Simple Incremental/Assets/Scripts/Monobehaviours/ObjectPooler.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System.Collections.Generic;
+using UnityEditor;
+using UnityEngine;
+
+/* This class is used for pooling objects and reusing objects for performance
+ * This works best when lots of the same object are spawned multiple times
+ * such as projectiles from a gun */
+public class ObjectPooler : MonoBehaviour
+{
+    public static ObjectPooler instance;
+    public Dictionary<string, Queue<GameObject>> dict = null;
+
+    void Awake()
+    {
+        if (instance == null)
+        {
+            instance = this;
+            dict = new Dictionary<string, Queue<GameObject>>();
```

Comment:
Defining the dictionary in awake just for best practice. This is to prevent extremely hard-to-debug OnEnable issues if we were to ever extend this class later on.